File: Js/Lucian_s_Luscious_Lasagna/lasagna.js
const PREPARATION_MINUTES_PER_LAYER = 2;
const EXPECTED_MINUTES_IN_OVEN = 40;

function remainingMinutesInOven(actualMinutesInOven) {
    return EXPECTED_MINUTES_IN_OVEN - actualMinutesInOven;
}

function preparationTimeInMinutes(numberOfLayers) {
    return PREPARATION_MINUTES_PER_LAYER * numberOfLayers;
}

function totalTimeInMinutes(numberOfLayers, actualMinutesInOven) {
    return preparationTimeInMinutes(numberOfLayers) + actualMinutesInOven;
}


console.log(`remaining minutes in oven => ${remainingMinutesInOven(30)}`);
console.log(`preparation time in minutes => ${preparationTimeInMinutes(2)}`);
console.log(`total Time In Minutes => ${totalTimeInMinutes(3, 20)}`);

File: C_Sharp/Lucians_Luscious_Lasagna/models/Lasagna.cs
namespace Lucians_Luscious_Lasagna.Models;

public class Lasagna
{
    public int ExpectedMinutesInOven()
        => 40;

    public int RemainingMinutesInOven(int actualMinutes)
        => ExpectedMinutesInOven() - actualMinutes;

    public int PreparationTimeInMinutes(int layers)
        => layers * 2;

    public int ElapsedTimeInMinutes(int layers, int minutes)
        => PreparationTimeInMinutes(layers) + minutes;

}
